File: programs/rand.py
import random

[BOARDSIZE, EMPTY, BLACK, WHITE] = [15, 0, 1, 2]


def user(board, stone, time):
    i, j = random.randint(0, 14), random.randint(0, 14)
    while board[i][j] != EMPTY:
        i, j = random.randint(0, 14), random.randint(0, 14)
    return i, j


def main():
    splits = input().split(", ")
    board = [[]] * 15
    for row in range(15):
        board[row] = [0] * 15
    for i in range(15):
        for j in range(15):
            board[i][j] = int(splits[i*15+j])

    stone = int(splits[225])
    time = float(splits[226])
    i, j = user(board, stone, time)
    print(i, j)


if __name__ == '__main__':
    main()

File: programs/rand.rs
use std::io;
use rand::Rng;

fn main() {
    let mut buffer = String::new();
    io::stdin().read_line(&mut buffer).unwrap();
    let mut splits = buffer.split(", ");

    let mut board = vec![0; 225];
    let mut stone = 0;
    let mut time = 0.0;

    for i in 0..225 {
        board[i] = splits.next().unwrap().parse::<i32>().unwrap();
    }
    stone = splits.next().unwrap().parse::<i32>().unwrap();
    time = splits.next().unwrap().parse::<f64>().unwrap();

    // eprintln!("{:?} {} {}", board, stone, time);

    let (x, y) = run(board, stone, time);
    println!("{} {}", x, y);
}


fn run(board: Vec<i32>, stone: i32, time: f64) -> (usize, usize) {
    let [mut x, mut y] = [0, 0];

    let mut rng = rand::thread_rng();

    while board[y * 15 + x] != 0 {
        x = rng.gen_range(0..15);
        y = rng.gen_range(0..15);
    }

    return (x, y);
}
